src/core_ocean/analysis_members/mpas_ocn_moc_streamfunction.F

```diff
@@ -559,6 +559,113 @@ subroutine ocn_compute_moc_streamfunction(domain, timeLevel, err)!{{{
+                       * 0.5_RKIND*(layerThickness(k,c1) + layerThickness(k,c2))
+                 end do
+              end do
+              do k = 2, nVertLevels
+                 mocStreamValLatAndDepthRegionLocal(1, k, iTransect) = &
+                    mocStreamValLatAndDepthRegionLocal(1, k - 1, iTransect) &
+                    + sumTransport(k - 1, iTransect)
+              end do
+           end do
+
+           !!!! END TRANSECT CALCULATION
+
+           do iCell = 1,nCellsSolve
+              iBin = MAX(int((latCell(iCell) - binBoundaryMocStreamfunction(1)) / binWidth), 2)
+              do k = 1,maxLevelCell(iCell)
+                 do i = 1, regionsInAddGroup
+                    currentRegion = regionsInGroup(i, regionGroupNumber)
+                    sumVertBinVelocityRegion(iBin, k, i) = sumVertBinVelocityRegion(iBin, k, i) + (vertVelocityTop(k, iCell) * &
+                          areaCell(iCell) * regionCellMasks(currentRegion, iCell))
+                 end do
+                 sumVertBinVelocity(iBin, k) = sumVertBinVelocity(iBin, k) + (vertVelocityTop(k, iCell) * areaCell(iCell))
+              end do
+           end do
+
+           do i = 1, regionsInAddGroup
+              do k = 1,nVertLevels
+                 do iBin = 2, nMocStreamfunctionBinsUsed + 1
+                    mocStreamValLatAndDepthLocal(iBin, k) = mocStreamValLatAndDepthLocal(iBin-1, k) &
+                       + sumVertBinVelocity(iBin, k)
+                    mocStreamValLatAndDepthRegionLocal(iBin, k, i) = mocStreamValLatAndDepthRegionLocal(iBin-1, k, i) &
+                       + sumVertBinVelocityRegion(iBin, k, i)
+                 end do
+              end do
+           end do
+
+           block => block % next
+       end do
+
+       call mpas_dmpar_sum_real_array(dminfo, nVertLevels * (nMocStreamfunctionBinsUsed + 1), mocStreamValLatAndDepthLocal, &
+           mocStreamvalLatAndDepthTotal)
+
+       call mpas_dmpar_sum_real_array(dminfo, nVertLevels * (nMocStreamfunctionBinsUsed + 1) * maxRegionsInGroup, &
+           mocStreamValLatAndDepthRegionLocal, mocStreamvalLatAndDepthRegionTotal)
+
+       call mpas_pool_get_subpool(domain % blocklist % structs, 'mocStreamfunctionAM', mocStreamfunctionAMPool)
+       call mpas_pool_get_array(mocStreamfunctionAMPool, 'mocStreamvalLatAndDepthGM', mocStreamvalLatAndDepthGM)
+       mocStreamvalLatAndDepthGM = mocStreamvalLatAndDepthTotal * m3ps_to_Sv
+
+       call mpas_pool_get_array(mocStreamfunctionAMPool, 'mocStreamvalLatAndDepthRegionGM', mocStreamvalLatAndDepthRegionGM)
+       mocStreamvalLatAndDepthRegionGM = mocStreamvalLatAndDepthRegionTotal * m3ps_to_Sv
+
+       !Add GM bolus contribution to resolved streamfunction to create total streamfunction
+       mocStreamvalLatAndDepthRegion = mocStreamvalLatAndDepthRegion + mocStreamvalLatAndDepthRegionGM
+       mocStreamvalLatAndDepth = mocStreamvalLatAndDepth + mocStreamvalLatAndDepthGM
```

Comment:
@milenaveneziani I've added these two lines to create a total overturning stream function

Reply:
great, thanks @vanroekel